packages/core/src/common/options.ts

```diff
@@ -103,11 +103,18 @@ export const buildOptions = ({ cliOptions }: { cliOptions: CliOptions }) => {
 
     factoryAddressCountThreshold: 1_000,
 
-    // os.freemem() / 4, bucketed closest to 64, 128, 256, 512, 1024, 2048 mB
+    // v8.getHeapStatistics().heap_size_limit / 4, bucketed closest to 128, 256, 512, 1024, 2048 mB
```

Comment:
LGTM, though I might move this to it's own line above the return statement like we do with `port`.